src/backend/common/queries/database_query.py

```diff
@@ -13,12 +13,30 @@
 
 
 class DatabaseQuery(abc.ABC, Generic[QueryReturn]):
-    _query_args: int
+    _cache_key: str = ""
+    _query_args: Dict[str, Any]
+    DATABASE_QUERY_VERSION = 0
+    BASE_CACHE_KEY_FORMAT: str = (
+        "{}:{}:{}"  # (partial_cache_key, cache_version, database_query_version)
+    )
+    CACHE_KEY_FORMAT: str = ""
+    CACHE_VERSION: int = 0
     DICT_CONVERTER: Type[ConverterBase[QueryReturn]] = ConverterBase
 
     def __init__(self, *args, **kwargs):
         self._query_args = kwargs
 
+    @property
+    def cache_key(self) -> str:
+        if self._cache_key == "":
```

Comment:
Can we do `if not self._cache_key` here? Or is actually comparing to an empty string important? The other bit would be to just leave `_cache_key` as an uninitialized variable